src/OpenTelemetry.Exporter.Zipkin/Implementation/LocalFileStorage.cs

```diff
@@ -0,0 +1,290 @@
+﻿// <copyright file="LocalFileStorage.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+
+namespace OpenTelemetry.Exporter.Zipkin.Implementation
+{
+    internal class LocalFileStorage : IDisposable
+    {
+        private readonly string path;
+        private readonly long maxSize;
+        private readonly int maintenancePeriod;
+        private readonly long retentionPeriod;
+        private readonly int writeTimeout;
+        private readonly Thread maintenanceThread;
+        private readonly AutoResetEvent maintenanceTrigger = new AutoResetEvent(false);
+
+        internal LocalFileStorage(
+                                string path,
+                                long maxSize = 52428800,
+                                int maintenancePeriod = 60,
+                                long retentionPeriod = 172800,
+                                int writeTimeout = 60)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            this.path = this.CreateTelemetrySubdirectory(path);
```

Comment:
For abstraction, the storage class will have no idea whether it is used for telemetry or something else - the function should be renamed.